### ISSUE TITLE
Installation error

### ISSUE THREAD

**Issue by wgmao** [bnprks/BPCells#47]
I tried to install `BPCells` on a different machine and it failed this time with following error message
```
factorization/GLMFactorization.h:153:74:   required from here
/data/home/wm8285/R/x86_64-pc-linux-gnu-library/4.2/RcppEigen/include/Eigen/src/Core/CoreEvaluators.h:960:8: warning: ignoring attributes on template argument ‘Eigen::internal::packet_traits<double>::type’ {aka ‘__vector(2) double’} [-Wignored-attributes]
make: *** [factorization/GLMFactorization.o] Error 1
ERROR: compilation failed for package ‘BPCells’
* removing ‘/data/home/wm8285/R/x86_64-pc-linux-gnu-library/4.2/BPCells’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmplGK56U/file4947546c0202/BPCells_0.1.0.tar.gz’ had non-zero exit status
```
Could you help resolve this error? Thanks!

**Comment by bnprks** [comment]
The chunk of the error message you shared is just a little bit too short for me to see what's going wrong. Could you copy-paste the whole error into a collapsible section following [these instructions](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections)? 

I assume from the file paths that you're building on Linux, but let me know if there are any differences you think are relevant compared to the machine you've successfully installed on

**Comment by wgmao** [comment]
Sorry about the incompleteness. Please see the information below.

<details>

<summary>Error message</summary>

[BP.log](https://github.com/bnprks/BPCells/files/12623144/BP.log)


</details>

<details>
<summary>Makevars</summary>
```
PKG_CPPFLAGS = -I '/data/home/wm8285/miniconda3/include/rlang' -I '../inst/include' -I '/data/home/wm8285/miniconda3/include' -I '/data/home/wm8285/lib' -L '/data/home/wm8285/miniconda3/lib'
PKG_CFLAGS = -I '/data/home/wm8285/miniconda3/include/rlang' -I '../inst/include' -I '/data/home/wm8285/miniconda3/include' -I '/data/home/wm8285/lib' -L '/data/home/wm8285/miniconda3/lib'
PKG_CXXFLAGS = -I '../inst/include' -I '/data/home/wm8285/miniconda3/include' -I '/data/home/wm8285/lib' -L '/data/home/wm8285/miniconda3/lib'
PKG_CCFLAGS = -I '../inst/include' -I '/data/home/wm8285/miniconda3/include' -I '/data/home/wm8285/lib' -L '/data/home/wm8285/miniconda3/lib'
LDFLAGS = -I /data/home/wm8285/miniconda3/include
CXX14 = g++ -std=gnu++14 -fPIC -g -O2
CXX18 = g++ -std=gnu++14 -fPIC -g -O2
CXX = g++ -std=gnu++17
```
</details>

<details>
<summary>sessionInfo</summary>
```
R version 4.3.1 (2023-06-16)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /data/home/wm8285/miniconda3/envs/cytospace/lib/libopenblasp-r0.3.21.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
[1] compiler_4.3.1 tools_4.3.1    curl_5.0.1     remotes_2.4.2
```
</details>

**Comment by bnprks** [comment]
Thanks for the extra info! I think this should be fixed with an update I just pushed to the main branch, but do let me know if the problem continues for you.

I'm not sure why exactly your particular installation hit this error when my installation attempts haven't hit it. I think this was due to slightly incorrect C++ library includes in a piece of the code that's currently unused. So I've disabled the code and fixed the import. Your install log gave the extra info I needed to be able to google an answer, so thanks for that.
